samples/cpp/benchmark_app/utils.cpp

```diff
@@ -132,32 +146,100 @@ void parse_value_for_virtual_device(const std::string& device, std::map<std::str
+            // Just keep the element that the key is virtual device AUTO/HETERO
             // e.g. AUTO:xxx,xxx -nstreams 2 will trigger exception that AUTO plugin didn't support nstream property.
             auto value = item_virtual->second;
             values_string.clear();
             values_string[device] = value;
             return;
         }
     }
+    std::stringstream ss;
     auto iter = values_string.begin();
     while (iter != values_string.end()) {
         if (iter->first == device) {
             iter++;
             continue;
         }
-        values_string[device] += iter->first + " " + iter->second + " ";
+        if (ss.str().empty())
+            ss << '{';
+        else
+            ss << ',';
+        ss << iter->first << ":" << iter->second;
         iter = values_string.erase(iter);
     }
-    if (values_string.find(device) != values_string.end()) {
-        auto& nstreams = values_string[device];
-        // Remove the space at the tail.
-        nstreams.pop_back();
-    }
+    if (!ss.str().empty())
+        ss << '}';
+    values_string[device] = ss.str();
     return;
 }
 
+template <typename T>
+void update_device_config_for_virtual_device(const std::string& value,
+                                             ov::AnyMap& device_config,
+                                             ov::Property<T, ov::PropertyMutability::RW> property,
+                                             std::map<std::string, bool>& is_dev_set_property,
+                                             bool is_load_config) {
+    // check if the element contains the hardware device property
+    if (split(value, ':').size() == 1) {
+        device_config[property.name()] = value;
+    } else {
+        // set device nstreams properties in the AUTO/MULTI/HETERO plugin
+        std::stringstream strm(value);
+        std::map<std::string, std::string> devices_property;
+        ov::util::Read<std::map<std::string, std::string>>{}(strm, devices_property);
+        for (const auto& it : devices_property) {
+            const auto& device_name = it.first;
+            const auto& device_value = it.second;
+            if (device_config.find(ov::device::properties.name()) == device_config.end() ||
+                (is_load_config && is_dev_set_property[device_name])) {
+                // Create ov::device::properties with ov::num_stream/ov::inference_precision and
+                // 1. Insert this ov::device::properties into device config if this
+                // ov::device::properties isn't existed. Otherwise,
+                // 2. Replace the existed ov::device::properties within device config.
+                is_dev_set_property[device_name] = false;
+                device_config.erase(device_name);
+                device_config[ov::device::properties.name()] = ov::AnyMap{};
+                auto& secondary_property = device_config.at(ov::device::properties.name()).as<ov::AnyMap>();
```

Comment:
Need to update the function `load_config()` to handle the loading secondary properties from config file, and replace `"DEVICE_PROPERTIES"` with `ov::device::properties.name()`.
`config[deviceName][hw_device_name] = hw_device_properties;` -> `config[deviceName][ov::device::properties.name()][hw_device_name] = hw_device_properties;`

https://github.com/openvinotoolkit/openvino/blob/b624b9806c090689c420977c6479c65b06b28706/samples/cpp/benchmark_app/utils.cpp#L835

Reply:
Good catch. Now dumping and loading can be significantly simplified.